### ISSUE TITLE
After binding OBS to a network adapter, NDI output doesn't bind to that adapter

### ISSUE THREAD

**Issue by SenselessData** [DistroAV/DistroAV#26]
PC with OBS has two IP addresses:

192.168.0.127
192.168.1.14

After binding OBS in "Bind To Interface" to 192.168.0.127 and enabling the OBS Output, the output only appears on the 192.168.1.14 network.

**Comment by Palakis** [comment]
Yes, currently those are two differents things in OBS. Getting the selected adapter from OBS settings is possible, but I need to check if I can pass this information to the NDI library.

**Comment by Palakis** [comment]
Doesn't seem possible. Btw here's an excerpt from the documentation of the NDI SDK version I use :

> If using more than a few senders and receivers and your system has more than one NIC then it is worth
connecting all available ports to the network; you will see bandwidth being distributed across multiple
network adapters (requires NDI 2.0 or better).

**Comment by SenselessData** [comment]
Thanks for the reply. We've got a NewTek technician on site today so I'll speak to them to see if they can think of a solution within the NDI SDK or otherwise.

**Comment by Palakis** [comment]
@SenselessData Btw, with which kind of hardware/software NDI source do you use the plugin with?

**Comment by SenselessData** [comment]
@Palakis We're currently just using Display Capture on OBS, then using the plugin to push the video onto our network with a NewTek VMC1 trying to view the video.

**Comment by teganscott** [comment]
Hi, 

I am having the same issue with the latest version of OBS and this NDI plugin. How can I select what network the NDI stream will go to? I tried the Binding IP, but see here that it doesn't have an effect. I've tried setting the preferred network to be the primary one, but that doesn't seem to help either. I need this solution to work both on Mac and PC, so I don't think NewTek's ForceBind Application will do the job as it is Windows only.

The help is much appreciated, thank you!

**Comment by jammerxd** [comment]
I also need this feature. So far the only way to get this to work is to disable the primary ethernet adapter on one of the machines(I have a gaming pc and a stream pc with a dedicated nic running between them) and then add the NDI capture so it's forced to use the dedicated nics on the machines. A bind to IP/Interface or even honoring the one OBS has built in is VERY MUCH NEEDED.

**Comment by teganscott** [comment]
Hey @jammerxd 
I didn't want to respond yet until I knew for sure it was working, but in early tests it looks like the best route to go is using NDI's Access Manager. It is an application from NewTek. On my outgoing computer in the Output I added a new Group with a custom name, say for example "NDIGrp1", and deleted the public group. Then on my incoming computer, I added the same group name to the "Recieve Groups" and the IP addressed associated to the outgoing PC in 'Remote Sources.' This stops NDI using the discovery tool which flip flops around IP addresses, and instead forces it to go through the IP on the computer the group is linked to. This is separate from OBS, but seems to work well with OBS's NDI stream on both Windows and Mac

**Comment by jammerxd** [comment]
Does it work with the OBS NDI plugin? @teganscott

**Comment by jammerxd** [comment]
Can you link me to this utility?

**Comment by teganscott** [comment]
So far it seems to be working with the OBS plugin
https://www.newtek.com/ndi/tools/

**Comment by jammerxd** [comment]
I am unable to download anything and I get the error that "Unable To Display Form" "Sorry, a browser extension you are using is preventing the form from displaying. Please disable it and try again." I tried in edge and firefox that don't have any extensions, even tried on my smart phone off of my wireless(I have heavy ad blocking on my local network to stop spam/ads) and it still wouldn't let me download them.

**Comment by teganscott** [comment]
Hey @jammerxd 
So bad news, the Access Manager doesn't seem to work with more rigorous tests. It could still be useful in some situations. In mine it doesn't, the stream still flips flops between networks.  
Let me know if you find a solution that works for you!

**Comment by jammerxd** [comment]
@teganscott I'll run some more tests myself but so far it seemed to follow what I set in the access manager.

**Comment by Palakis** [comment]
https://github.com/Palakis/obs-ndi/pull/313 should help with that.

**Comment by MeDott29** [comment]
> Hey @jammerxd
> I didn't want to respond yet until I knew for sure it was working, but in early tests it looks like the best route to go is using NDI's Access Manager. It is an application from NewTek. On my outgoing computer in the Output I added a new Group with a custom name, say for example "NDIGrp1", and deleted the public group. Then on my incoming computer, I added the same group name to the "Recieve Groups" and the IP addressed associated to the outgoing PC in 'Remote Sources.' This stops NDI using the discovery tool which flip flops around IP addresses, and instead forces it to go through the IP on the computer the group is linked to. This is separate from OBS, but seems to work well with OBS's NDI stream on both Windows and Mac

This solution has solved my problem.  Thank you.  I spent hours trying to fix this.

**Comment by crashworldtk** [comment]
I’m having this issue on my Linux setup, the OBS-NDI plugin grabs the wrong ip address, not the default gateway

**Comment by pinpox** [comment]
Having the same issue still, not sure why this was closed. I'm streaming from my phone to my notebook, the notebook is connected to wifi and wired, while the phone is only connected to the wifi. 

This gives me a black screen, even though I'm able to find the source, until I disable the notebooks wired connection.
It would be very helpful to be able to use the wired connection for streaming while receiving via the wifi.

**Comment by paulpv** [comment]
I am also not sure why this was closed in 2017.
tt2468 addressed this in his rewrite branch, so I can look at that and see what I can port over to the current branch.

**Comment by pinpox** [comment]
> I can look at that and see what I can port over to the current branch.

That would be awesome! Let me know if you need testing

**Comment by Trouffman** [comment]
> Having the same issue still, not sure why this was closed. I'm streaming from my phone to my notebook, the notebook is connected to wifi and wired, while the phone is only connected to the wifi. 
> 
> This gives me a black screen, even though I'm able to find the source, until I disable the notebooks wired connection.
> It would be very helpful to be able to use the wired connection for streaming while receiving via the wifi.

The reason you can find the source but not get the payload (stream) from it is because the discovery is not (AFAIK) linked to the data stream.

In short :
- You can see the stream listed (aka NDI Discovery)
- No payload (stream is black), because it is trying to connect to the stream on the wifi (ip provided by the Discovery), but accessing it via the wired network.


For this you need to check :
- Your network routes for both Interface 
WiFi goes to only the WiFi ip range 
Wired to everything else (for example)

- list the following IPs
OBS PC wifi
OBS PC wired
Phone (WiFi)

- what's your route list 
`route print` on windows.
`ip route show` on Linux/unix

**Comment by Trouffman** [comment]
Also this was planned as part of the "rewrite" branch (for the initial issue logged).

**Comment by Jonny4455** [comment]
Did anybody find a solution to this? The only thing i found that kinds worked was to disable the adapter you don't want to use for Ndi. Then start obs and it will go through the adapter you want. Then just re-enable the other adapter and your good

**Comment by RayneYoruka** [comment]
> Did anybody find a solution to this? The only thing i found that kinda worked was to disable the adapter you don't want to use for Ndi. Then start obs and it will go through the adapter you want. Then just re-enable the other adapter and your good

This is what you need https://docs.ndi.video/docs/white-paper/nic-selection

**Comment by pinpox** [comment]
Was this fixed or just closed?

**Comment by RayneYoruka** [comment]
What is there to fix? Create a new bug if you encounter issues.
